ExtraDry/ExtraDry.Blazor/Components/Standard/TriSwitch.razor

```diff
@@ -0,0 +1,10 @@
+﻿@namespace ExtraDry.Blazor
+@implements IExtraDryComponent
+
+<div class="@CssClasses" @attributes="@UnmatchedAttributes">
+    <input id="@Id" type="checkbox" checked="@Switched" @onchange="DoChange" @onclick="OnClicked" />
```

Comment:
Is there a benefit to leaving the backing field as a <input type=checkbox /> ?  This forces you to use JavaScript interop which is kludgy.  It might allow for some better keyboard navigation or similar.

Reply:
I'm not sure I follow the benefits of making the field nullable apart from not having to use JavaScript interop.

I've had a go at trying to replicate the behaviour without JavaScript but I couldn't get the indeterminate state to set correctly (which is probably down to my knowlege gap).

Reply:
Sorry, that question mark was actually me putting in HTML which must have been escaped.  I originally meant as an 'input' field with type equal to 'checkbox'.  That is, the rendering is no longer as a checkbox and is completely replaced by HTML.  Probably easier to style without it being a checkbox.  Also don't need to JavaScript interop.  However, keyboard navigation and shortcuts might not work as well.

Reply:
Ok, thinking about it a bit more, it does make sense for it to semantically be a input control.  It _is_, after all, providing input.  Just having a look through the options though, it seems the only alternatives are a select tag with three values, or a input tag of type range with values {-1, 0, 1} mapping to "No", "Not Answered", and "Yes".  Neither is awesome, so maybe a tri-state checkbox is as good as it gets.

Reply:
The input checkbox has been removed and the state is being handled by Blazor.  This removed the need for the JavaScript interop.

Keyboard navigation has been added in.